renku/service/serializers/headers.py

```diff
@@ -32,42 +36,108 @@ def decode_b64(value):
         return value
 
 
+def encode_b64(value):
+    """Encode value to base64."""
+    if isinstance(value, str):
+        value = bytes(value, "utf-8")
+
+    return base64.b64encode(value).decode("utf-8")
+
+
+class UserIdentityToken(Schema):
+    """User identity token schema."""
+
+    jti = fields.String()
+    exp = fields.Integer()
+    nbf = fields.Integer()
+    iat = fields.Integer()
+    iss = fields.String()
+    aud = fields.List(fields.String())
+    sub = fields.String()
+    typ = fields.String()
+    azp = fields.String()
+    nonce = fields.String()
+    auth_time = fields.Integer()
+    session_state = fields.String()
+    acr = fields.String()
+    email_verified = fields.Boolean()
+    preferred_username = fields.String()
+    given_name = fields.String()
+    family_name = fields.String()
+
+    email = fields.String(required=True)
+    name = fields.String(required=True)
+    user_id = fields.String()  # INFO: Generated post load.
+
+    @post_load
+    def set_user_id(self, data, **kwargs):
+        """Sets users id."""
+        data["user_id"] = encode_b64(secure_filename(data["email"]))
+        return data
+
+
 class UserIdentityHeaders(Schema):
     """User identity schema."""
 
-    user_id = fields.String(required=True, data_key="renku-user-id")
-    fullname = fields.String(data_key="renku-user-fullname")
-    email = fields.String(data_key="renku-user-email")
-    token = fields.String(data_key="authorization")
+    user_token = fields.String(required=True, data_key="renku-user")
+    auth_token = fields.String(required=True, data_key="authorization")
 
-    def extract_token(self, data):
-        """Extract token."""
-        value = data.get("authorization", "")
-        components = value.split(" ")
+    @staticmethod
+    def decode_token(token):
+        """Extract authorization token."""
+        components = token.split(" ")
 
-        rfc_compliant = value.lower().startswith("bearer")
+        rfc_compliant = token.lower().startswith("bearer")
         rfc_compliant &= len(components) == 2
 
         if not rfc_compliant:
             raise ValidationError("authorization value contains invalid value")
 
         return components[-1]
 
-    @pre_load()
+    @staticmethod
+    def decode_user(data):
+        """Extract renku user from a JWT."""
+        decoded = jwt.decode(data, JWT_SECRET_TOKEN, algorithms=["HS256"], audience="renku",)
```

Comment:
This could raise `ExpiredSignatureError`, `ImmatureSignatureError`, `InvalidIssuedAtError` , should we handle that?